x/authz/keeper/keeper_test.go

```diff
@@ -370,7 +370,7 @@ func (s *TestSuite) TestDequeueAllGrantsQueue() {
 	require.NoError(err)
 
 	newCtx := s.ctx.WithBlockTime(exp.AddDate(1, 0, 0))
-	err = s.authzKeeper.DequeueAndDeleteExpiredGrants(newCtx)
+	err = s.authzKeeper.DequeueAndDeleteExpiredGrants(newCtx, 200)
```

Comment:
Is this testing the new change?

Reply:
@mpoke Added tests and refactored code.

While testing I refactored the code so that you can set the limit to be 0 pruned records. The previous implementation would prune a minumum of 1 record.

This is just an edgecase.